slo_lambda/src/windowed_slo.py

```diff
@@ -76,164 +56,74 @@ class SingleMetric(MetricBase):
       }
     """
 
-    def __init__(self, namespace: str, metric_name: str, dimensions: List):
+    def __init__(
+        self,
+        window_days: int,
+        namespace: str,
+        metric_name: str,
+        dimensions: List,
+        statistic: Optional[str] = "Sum",
+        extended_statistic: Optional[str] = None,
+        multiplier: float = 1.0,
+    ):
         self.namespace = namespace
         self.metric_name = metric_name
         self.dimensions = dimensions
+        self.statistic = statistic
+        self.extended_statistic = extended_statistic
+        self.multiplier = float(multiplier)
+
         self.stat_args = {
-            'StartTime': datetime.datetime.utcnow() - datetime.timedelta(days=WINDOW_DAYS),
-            'EndTime': datetime.datetime.utcnow(),
-            'Period': WINDOW_DAYS * 24 * 60 * 60,
-            'Statistics': [
-                'Sum',
-            ],
+            "StartTime": datetime.datetime.utcnow()
+            - datetime.timedelta(days=window_days),
+            "EndTime": datetime.datetime.utcnow(),
+            "Period": window_days * 24 * 60 * 60,
         }
+        if extended_statistic:
+            self.stat_args["ExtendedStatistics"] = [extended_statistic]
+        else:
+            self.stat_args["Statistics"] = [statistic]
 
     def sum(self) -> float:
         """
-        sum returns the sum of a single cloudwatch metric over WINDOW_DAYS
+        sum returns the sum of a single cloudwatch metric over window_days
+        multiplied by the multiplier.
         """
         total = 0.0
         for datapoint in Cloudwatch.client().get_metric_statistics(
-                Namespace=self.namespace,
-                MetricName=self.metric_name,
-                Dimensions=self.dimensions,
-                **self.stat_args,
-        )['Datapoints']:
+            Namespace=self.namespace,
+            MetricName=self.metric_name,
+            Dimensions=self.dimensions,
+            **self.stat_args,
+        )["Datapoints"]:
             total += self.extract_stat(datapoint)
-        return total
+        return total * self.multiplier
```

Comment:
@akrito - And here is the only other part needed for the multiplier.

With a multiplier of -1 the output will be subtracted from other values in the composite.

Reply:
sweet